### DISCUSSION TITLE
Phase out native Windows Meteor development tools in favour of WSL2/Ubuntu

### DISCUSSION THREAD

**Discussion by vlasky** [meteor/meteor#12648] [Feature Requests]
I run Windows 10. Ever since I switched to carrying out Meteor development within WSL2/Ubuntu several years ago, I have been very happy and never looked back.

Meteor builds apps much, much faster within WSL2 than under Windows. Also, the Windows version of VS Code has native support for editing files within the WSL2 Ubuntu environment, so you get the benefits of both worlds.

Given this, it's my personal opinion that the native Windows Meteor tools should be phased out. I and other developers whom I have collaborated with have had issues with the Windows tools at one time or another, requiring reinstallation or other troubleshooting.

I feel that the slow native Windows build speeds cause developer frustration and can only be creating unnecessary bad PR for Meteor. 

What are your thoughts?

**Comment by StorytellerCZ** [comment]
I agree. We should probably start with the docs to guide people towards WSL2. PRs welcome!

**Comment by wreiske** [comment]
I'm interested in making this work on Windows without WSL. I have approval from my employer to set a $500 bounty to improve the development experience on Windows.

I spend my time between my Macbook Pro, a remote ssh linux session, and local development on Windows 11. My biggest headache comes when it's time to `meteor update`. Everywhere else it just works. On Windows I get stuck for hours in "Selecting package versions" and it never seems to work the first time. I always have to Ctrl-C and try again, wait again, and then eventually it will finish.

@StorytellerCZ , who do you think would be best suited in the community (maybe even you!) to take on this challenge? If I had the free time I would dive in myself... I'm just swamped right now.

**Reply by StorytellerCZ** [reply]
Recently I have tried to get our designers to run Meteor on Windows and WSL. It did not go well, not because of Meteor, but due to setting up WSL. I will have to look on my own Windows setup (maybe today). But I'm not the guy to work on this as my primary OS is Ubuntu and I have way too much existing work to be able to give this justice. Not to mention I don't think I have the technical expertise to handle this.

That said, now if you create an issue, you can then go to https://console.algora.io/org/meteor to add a bounty on that issue.

**Reply by wreiske** [reply]
Bug bounty posted! https://github.com/meteor/meteor/issues/12935

**Comment by trusktr** [comment]
This might not be a good idea.

Some companies will not enable WSL due their security policies. It would be a lot better to continue to support the most out-of-the-box experience possible, working in a vanilla Windows install with as least friction as possible.

Besides that, some Windows developers will find it troubling to have to learn Linux if they already know PowerShell, etc.
